### PR TITLE
+ Add flickr oauth

### PR DESCRIPTION
**Pull request by williamchong007**
also added oauth page for generic oauth use
also refactor some social api flow

### CODE REVIEW

File: pages/in/oauth/_platform/_id/index.vue [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                ...
+              </span>
+              <span v-else>{{ $t('Verify.label.verifying') }}</span>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
```
Comment (nnkken) [inline pages/in/oauth/_platform/_id/index.vue]: Is this multi-level nesting and `...` intended and temporary?

File: pages/in/oauth/_platform/_id/index.vue [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            platform: 'flickr',
+            payload: {
+              oAuthToken: this.$route.query.oauth_token,
+              oAuthVerifier: this.$route.query.oauth_verifier,
+              user: this.username,
+            },
+          });
+          break;
+        }
+        default:
```
Comment (nnkken) [inline pages/in/oauth/_platform/_id/index.vue]: no Facebook?

File: pages/in/oauth/_platform/_id/index.vue (hunk 1 above)
Comment (williamchong007) [reply pages/in/oauth/_platform/_id/index.vue]: @nwingt i copied this from `/verify`, not sure if needed for layout

File: pages/in/oauth/_platform/_id/index.vue (hunk 2 above)
Comment (williamchong007) [reply pages/in/oauth/_platform/_id/index.vue]: no, since it does not require callback page if used with facebook SDK

**Pull request merged**